vcr/credential/builder.go

```diff
@@ -0,0 +1,73 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nuts-foundation/go-did"
+)
+
+const defaultContext = "https://www.w3.org/2018/credentials/v1"
+const nutsContext = "https://nuts.nl/credentials/v1"
+const defaultType = "VerifiableCredential"
+
+// Builder is an abstraction for extending a partial VC into a fully valid VC
+type Builder interface {
+	// Type returns the matching Verifiable Credential type
+	Type() string
+	// Build sets the defaults for common fields and creates the signature
```

Comment:
" and creates the signature" << that's a lie (according to the implementation). Now it comes to me; we're not creating signatures yet as part of this PR?

Also; I'd expect `Builder.Build()` to return the stuff that was built?

Reply:
Old comments I guess. Will fix.